app/helpers/pdf_helper.rb

```diff
@@ -34,4 +34,10 @@ def output_file
     PdfForms.new.fill_form(source_pdf_path, pdf_tempfile.path, hash_for_pdf)
     pdf_tempfile
   end
+
+  def pdf_mask(string, unmasked_char_count = 0)
+    return string unless string.present?
+
+    string.gsub(/.(?=.{#{unmasked_char_count}})/, 'X')
```

Comment:
It'd be easier to understand if this were implemented with `.slice` e.g. `masked = string.slice(-unmasked_char_count...)` plus adding the "X"s back to the start.

We attempted to use https://regex101.com/ to find out exactly how the regex works, but we weren't able to get very far that way; we did test it manually in `rails c` and it seems to do the right thing.

Not a blocker.